### DISCUSSION TITLE
LDAP mTLS, client TLS configuration (KC 19)

### DISCUSSION THREAD

**Discussion by danj3** [keycloak/keycloak#16588] [Q&A]
I'm trying to coax KC into providing client certs to my ldap server. I've followed the [outgoing HTTP](https://www.keycloak.org/server/outgoinghttp) instructions. These don't make any mention of LDAP as a consumer, but there is no other info on LDAP client certs.

Using keytool to build the keystore with a p12 from our internal CA, the only question that came up (and the only one in the whole process) was the `-alias`, does it need to be something specific for KC to find it.

The start command looks like:
```
/opt/keycloak/bin/kc.sh \                                                                                                                
        start \                                                                                                                                       
        --optimized \                                                                                                                                 
        --spi-connections-http-client-default-client-keystore=/opt/keycloak/conf/client.jks \                                                         
        --spi-connections-http-client-default-client-keystore-password=password \                                                                     
        --spi-connections-http-client-default-client-key-password=password \
       "$@"
```

I see no messages either way about this configuration being used or incorrect. Not sure where to go from here, any guidance would be profoundly great.

**Comment by danj3** [comment]
Any thoughts on this question? I've made no further discoveries.

**Comment by pravsjv** [comment]
@danj3 were you able to setup LDAP in MTLS with keycloak? If so could you share the details. Thanks!!

**Comment by danj3** [comment]
i have not been successful. i intend to try again with kc22 and hope for a
better outcome but am not optimistic.

**Comment by rajatkexternal** [comment]
@danj3 you are able to setup LDAP in mTLS with keycloak 22 ?

**Comment by danj3** [comment]
i have not tried in some time (versions), but now that i’m running 24 it
seems time to try again.

**Comment by tiborauer** [comment]
Any update on integrating LDAP with mTLS?
